File: apps/api/src/app/schemas/resume.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class Profile(BaseModel):
    name: str
    role: str
    age: str
    experience: str
    education: str
    phone: str
    email: str
    introduction: str
    avatar_url: str = ""
    location: str = ""
    availability: str = "开放机会"


class SkillGroup(BaseModel):
    title: str
    skills: list[str] = Field(default_factory=list)


class Experience(BaseModel):
    company: str
    role: str
    time: str
    content: list[str] = Field(default_factory=list)
    achievements: list[str] = Field(default_factory=list)


class Project(BaseModel):
    index: str
    name: str
    role: str
    time: str
    summary: str
    contribution: list[str] = Field(default_factory=list)
    stack: list[str] = Field(default_factory=list)


class AgentDemo(BaseModel):
    title: str
    description: str
    tags: list[str] = Field(default_factory=list)
    status: str
    demo_url: str = ""


class Expectation(BaseModel):
    roles: list[str] = Field(default_factory=list)
    location: str = ""
    salary: str = ""
    availability: str = ""


class ResumeData(BaseModel):
    profile: Profile
    strengths: list[str] = Field(default_factory=list)
    skill_groups: list[SkillGroup] = Field(default_factory=list)
    expectation: Expectation
    experiences: list[Experience] = Field(default_factory=list)
    projects: list[Project] = Field(default_factory=list)
    agent_demos: list[AgentDemo] = Field(default_factory=list)


class ResumeResponse(ResumeData):
    model_config = ConfigDict(from_attributes=True)

    updated_at: datetime | None = None

File: apps/web/src/lib/resume.ts
export type Profile = {
  name: string;
  role: string;
  age: string;
  experience: string;
  education: string;
  phone: string;
  email: string;
  introduction: string;
  avatar_url: string;
  location: string;
  availability: string;
};

export type SkillGroup = { title: string; skills: string[] };
export type Experience = { company: string; role: string; time: string; content: string[]; achievements: string[] };
export type Project = { index: string; name: string; role: string; time: string; summary: string; contribution: string[]; stack: string[] };
export type AgentDemo = { title: string; description: string; tags: string[]; status: string; demo_url: string };
export type Expectation = { roles: string[]; location: string; salary: string; availability: string };

export type ResumeData = {
  profile: Profile;
  strengths: string[];
  skill_groups: SkillGroup[];
  expectation: Expectation;
  experiences: Experience[];
  projects: Project[];
  agent_demos: AgentDemo[];
  updated_at?: string | null;
};

const publicApiBaseUrl =
  process.env.NEXT_PUBLIC_API_BASE_URL ?? "http://localhost:8000";
const apiBaseUrl =
  typeof window === "undefined"
    ? (process.env.API_INTERNAL_URL ?? publicApiBaseUrl)
    : publicApiBaseUrl;

export async function fetchResume(): Promise<ResumeData | null> {
  try {
    const response = await fetch(`${apiBaseUrl}/api/v1/resume`, {
      cache: "no-store",
      signal: AbortSignal.timeout(2500),
    });
    if (!response.ok) return null;
    return (await response.json()) as ResumeData;
  } catch {
    return null;
  }
}

export { apiBaseUrl };
